src/shell/shell_cmds.cpp

```diff
@@ -1450,6 +1450,21 @@ void DOS_Shell::CMD_TIME(char * args) {
 		mem_writed(BIOS_TIMER, static_cast<uint32_t>(ticks_now));
 		return;
 	}
+	uint32_t newhour, newminute, newsecond;
+	if (sscanf(args, "%u:%u:%u", &newhour, &newminute, &newsecond) == 3) {
+		if (newhour > 23 || newminute > 59 || newsecond > 59)
+			WriteOut(MSG_Get("SHELL_CMD_TIME_ERROR"));
+		else {
+			reg_ch = static_cast<Bit8u>(newhour);
+			reg_cl = static_cast<Bit8u>(newminute);
+			reg_dh = static_cast<Bit8u>(newsecond);
+
+			reg_ah=0x2d; // set system time
+			CALLBACK_RunRealInt(0x21);
+			if (reg_al == 0xff) WriteOut(MSG_Get("SHELL_CMD_TIME_ERROR"));
```

Comment:
Also minor white-space to put the condition and action on separate lines:
``` c++
if (reg_al == 0xff) {
	WriteOut(MSG_Get("SHELL_CMD_TIME_ERROR"));
}
```

Reply:
Fixed for both DATE and TIME commands.